pkg/provider/github/app/token.go

```diff
@@ -102,17 +102,16 @@ func (ip *Install) GetAndUpdateInstallationID(ctx context.Context) (string, stri
 	return enterpriseHost, token, installationID, nil
 }
 
-func (ip *Install) listRepos(ctx context.Context) (bool, error) {
-	if ip.repoList == nil {
-		var err error
-		ip.repoList, err = github.ListRepos(ctx, ip.ghClient)
-		if err != nil {
-			return false, err
-		}
+// matchRepos check for matching repo and update the instance repoList.
+func (ip *Install) matchRepos(ctx context.Context) (bool, error) {
+	installationRepoList, err := github.ListRepos(ctx, ip.ghClient)
+	if err != nil {
+		return false, err
 	}
-	for i := range ip.repoList {
+	ip.repoList = append(ip.repoList, installationRepoList...)
```

Comment:
will the installationRepoList is always unique or will there be any duplication ?

Reply:
IMO installationRepoList will always be unique. That is the list of repositories returned for an installation. Installation goes onto all or subset of repositories under single org or user, so they must be unique. Unless I'm missing something.